src/main/java/com/arpnetworking/tsdcore/model/RequestEntry.java

```diff
@@ -0,0 +1,109 @@
+/*
+ * Copyright 2020 Dropbox
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.arpnetworking.tsdcore.model;
+
+import com.arpnetworking.commons.builder.OvalBuilder;
+import net.sf.oval.constraint.NotNull;
+import org.asynchttpclient.Request;
+
+import java.util.Optional;
+
+/**
+ * Contains the info for a http request.
+ *
+ * @author Qinyan Li (lqy520s at hotmail dot com)
+ */
+public final class RequestEntry {
+    public Request getRequest() {
+        return _request;
+    }
+
+    public long getEnterTime() {
+        return _enterTime;
+    }
+
+    public Optional<Long> getPopulationSize() {
+        return _populationSize;
+    }
+
+    private RequestEntry(final Builder builder) {
+        _request = builder._request;
+        _enterTime = builder._enterTime;
+        _populationSize = builder._populationSize;
+    }
+
+    private final Request _request;
+    private final long _enterTime;
+    private final Optional<Long> _populationSize;
+
+    /**
+     * {@link com.arpnetworking.commons.builder.Builder} implementation for
+     * {@link RequestEntry}.
+     *
+     * TODO(ville): Convert RequestEntry.Builder would be a ThreadLocalBuilder
+     * See comments in HttpPostSink:createRequests
+     */
+    public static final class Builder extends OvalBuilder<RequestEntry> {
+
+        /**
+         * Public constructor.
+         */
+        public Builder() {
+            super(RequestEntry::new);
+        }
+
+        /**
+         * Set the request. Required. Cannot be null.
+         *
+         * @param value The request.
+         * @return This {@link Builder} instance.
+         */
+        public Builder setRequest(final Request value) {
+            _request = value;
+            return this;
+        }
+
+        /**
+         * Set the time when the request enter the pending request queue. Required. Cannot be null.
+         *
+         * @param value The enter time.
+         * @return This {@link Builder} instance.
+         */
+        public Builder setEnterTime(final long value) {
```

Comment:
A `long` is not really a time. At the very least the comment should specify how the long is used as a timestamp (e.g. millis since epoch). However, we should probably consider using a `java.time` construct instead. Especially, if we end up doing math on this. (I know MAD did it this way; a PR to update that similarly would be ideal, but I'll be happy with a GH issue).

Reply:
Changed and made PR in MAD (https://github.com/ArpNetworking/metrics-aggregator-daemon/pull/226/)

Reply:
Approved MAD PR conditional to getting the build to pass.